### PR TITLE
driver: smaract driver updates

### PR DESCRIPTION
**Pull request by amuskens**

### CODE REVIEW

File: src/odemis/driver/smaract.py [hunk 1]
```diff
@@ -1452,11 +1472,16 @@ def _doReference(self, future):
...
                 # wait till reference completes
                 while True:
-                    ev = self.WaitForEvent(MC_5DOF_DLL.SA_MC_INFINITE)
+                    ev = self.WaitForEvent(100)
                     # check if move is done
                     if ev.type == MC_5DOF_DLL.SA_MC_EVENT_MOVEMENT_FINISHED:
+                        logging.debug("Referencing finished")
+                        if ev.i32 != MC_5DOF_DLL.SA_MC_OK:
+                            logging.error("Referencing failed with error %s", ev.i32)
+                            raise IOError("Referencing failed with error %s" % (ev.i32,))
```
Comment (pieleric) [inline src/odemis/driver/smaract.py:1484]: I've noticed that ev.i32 contains a "normal" error code. So it should be possible to raise a `SA_MCError`

File: src/odemis/driver/smaract.py [hunk 2]
```diff
@@ -1046,7 +1046,7 @@ class MC_5DOF_DLL(CDLL):
 
     # property symbols
     SA_MC_PKEY_PIVOT_POINT_MODE = 0x00001001
```
Comment (pieleric) [inline src/odemis/driver/smaract.py:1048]: BTW, the include file has two extra properties:
```
#define SA_MC_PKEY_MODEL_CODE                           0x0a02
#define SA_MC_PKEY_MODEL_NAME                           0x0a03
```
The smaract_mc private bindings for python also mention a few extra keys, including 
`    VERSION_STRING                    = 0xf001`

Could you have a look at reading this properties, to update hwVersion and swVersion with useful info?

File: src/odemis/driver/smaract.py [hunk 3]
```diff
@@ -2830,6 +2878,47 @@ def _doReference(self, future, axes):
...
+                future._was_stopped = True
+                raise
+            except Exception:
+                logging.exception("Referencing failure")
+                raise
+            finally:
+                # We only notify after updating the position so that when a listener
+                # receives updates both values are already updated.
+                self._updatePosition()
+
```
[the remark this answers is not in the document]
Comment (amuskens) [reply src/odemis/driver/smaract.py:2923]: Should be fixed now in the other PR

**Comment by pieleric** [comment]
Could you also add an extra parameter to `MC_5DOF`, `hold_time`, like for the MCS2, to change the hold time.

**Pull request merged**